libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1722,9 +1723,7 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	TypeResult result = type(_operation.subExpression())->unaryOperatorResult(op);
 	if (!result)
 	{
-		string description = "Unary operator " + string(TokenTraits::toString(op)) + " cannot be applied to type " + subExprType->humanReadableName();
-		if (!result.message().empty())
-			description += ". " + result.message();
+		string description = "Unary operator " + string(TokenTraits::toString(op)) + " cannot be applied to type " + subExprType->humanReadableName() + (!result.message().empty() ? ". " + result.message() : ".");
```

Comment:
How about making the first full stop a fixed part of the message and the second full stop a part of the `result.message()`?
```suggestion
		string description = "Unary operator " + string(TokenTraits::toString(op)) + " cannot be applied to type " + subExprType->humanReadableName() + "." + (!result.message().empty() ? " " + result.message() : "");
```

Reply:
Looks like pretty much the same thing; will this make it easier for you to rebase? :smile:

Reply:
Looks similar but the difference is that the full stops do not depend on any logic. So you don't need to care about a full stop when you use `result.message()` somewhere else in code or when `result.message()` returns a string that already contains a full stop. Maybe, it would be a bit easier to rebase too :)

Reply:
I've made the changes in two places where it was applicable.